docs/common-sdk-requirements.md

```diff
@@ -0,0 +1,43 @@
+# Requirements for the Common SDK infrastructure
+The purpose of this doc is to have a place to list and discuss all requirements to the new modular SDK infrastructure (also informally called "the Common SDK infrastructure") before creating an official design.
+
+The information listed here was primarily sourced in a few meetings/discussions/whiteboardings we had before.
+
+## Terminology
+* _Common SDK_ is the new modular SDK. 
+* _Service_ is a single building unit (module) of the new modular SDK. TODO: please let me know if another term is desired for this.
+
+## Requirements
```

Comment:
Another requirement is that the Code Generated should feel as close to the platform/engine as possible. For example, while the design should be agnostic of language specifics -- the final implementation of core components and plugability model should follow familiar patterns in the language. For example, Interfaces might be the way to go in C#/Unity - but C++ might be suited to abstract base classes to work around ABI issues ( because the vtable format seldom differs among compilers) etc.

Reply:
Agreed. And honestly... Impossible to avoid. There's a certain amount of "that's just the only way to do it" that applies to many of these wildly different languages.